autoload/lsp/completion.vim

```diff
@@ -494,7 +494,7 @@ export def BufferInit(lspserver: dict<any>, bnr: number, ftype: string)
     endif
     # <Enter> in insert mode stops completion and inserts a <Enter>
     if !opt.lspOptions.noNewlineInCompletion
-      inoremap <expr> <buffer> <CR> pumvisible() ? "\<C-Y>\<CR>" : "\<CR>"
+      inoremap <expr> <buffer> <CR> pumvisible() ? "\<C-Y>" : "\<CR>"
```

Comment:
You can set the "noNewlineInCompletion" option to TRUE to not add the newline in the map.  If you remove the newline from the map, then it will not be inserted when pressing C-Y.

Reply:
That is fine. I want <C-Y> to **be** mapped. But when we have this mapping, hitting <CR> doesn't properly insert the completion, atleast for snippet completions:
![lsp_complete](https://user-images.githubusercontent.com/16995641/229278393-d7da7579-3394-44ec-992a-e3adc400066a.gif)

Hitting <C-Y> on snippet completion properly inserts it, but <CR> doesn't insert it. But with the above fix, it is properly inserted.